### PR TITLE
Add exp-4327 as well as switch to poetry.

### PR DESCRIPTION
**Pull request by jrbenny35**
Fixes https://mozilla-hub.atlassian.net/browse/EXP-4327

### CODE REVIEW

**Comment by jrbenny35** [comment]
Commented out some code for now, it is quite complex and I need to simplify it.

File: tests/conftest.py [hunk 1]
```diff
@@ -111,37 +118,37 @@ def firefox_options(
...
-        firefox_options.add_argument(setup_profile)
-    if request.config.getoption("--run-firefox-release"):
-        binary = Path("utilities/firefox-release/firefox/firefox-bin").absolute()
-        firefox_options.binary = f"{binary}"
-    if request.node.get_closest_marker("reuse_profile") and not request.config.getoption(
-        "--run-update-test"
-    ):
-        firefox_options.add_argument("-profile")
-        firefox_options.add_argument(setup_profile)
+    # if request.config.getoption("--run-update-test"):
```
Comment (pdehaan) [inline tests/conftest.py:121]: Is this the aforementioned complex code in need of simplifying?

File: tests/features/generic_nimbus.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+@generic_nimbus
+Feature: Generic Nimbus smoke tests all pass
+
+    @smoke
+    Scenario: The browser will enroll into the requested branch
+        Given Firefox is launched enrolled in an Experiment
+        Then The experiment branch should be correctly reported
```
Comment (pdehaan) [inline tests/features/generic_nimbus.feature:11]: Nit: add line break at End of file

File: tests/conftest.py (hunk 1 above)
Comment (jrbenny35) [reply tests/conftest.py:121]: Correct

**Pull request merged**